packages/@react-aria/combobox/test/useComboBox.test.js

```diff
@@ -65,4 +70,30 @@ describe('useComboBox', function () {
     expect(buttonProps['onPressStart']).toBeTruthy();
     expect(buttonProps['onKeyDown']).toBeTruthy();
   });
+
+  it('should prevent default on Enter if isOpen', function () {
```

Comment:
Would've like to do this as a component test, but had problems getting the form submit to behave like it does in browser (hitting Enter would either never cause onSubmit to fire or it would always cause it to fire depending on the fireEvent/userEvent)